### PR TITLE
Reimplement codespaces support

### PR DESCRIPTION
**Pull request by hazzadous**
This change:

 * separates the docker-compose.yml for codespaces completely. This is done to reduce complexity of trying to unify the two right now, although it looks pretty close to be able to merge the two. I'm going to consider that out of scope though for this PR
 * adds a dedicated image for codespaces, that includes more than the dev.Dockerfile image includes. I'm not entirely sure what dev.Dockerfile is used for but it seems from the comments in that file that it might be used for more than just local dev? Anyhow, again I'd like to consider any unification separate from this PR.
 * Add a github workflow to build the image used by codespaces

This gets us to something that we can then run `pytest ee` , there looks like there's more to get `yarn start-http` working, but these can be implemented separately as I or others use the setup.

This is [it almost working](https://www.loom.com/share/3aa89c35af33431db414f811c5da3073) as expected, aside for some django template test failure at the end

### CODE REVIEW

**Comment by hazzadous** [comment]
@guidoiaquinti @pauldambra I just yolo'd this in to make my life easier, but if you have feedback would love some

File: .devcontainer/container_start.sh [hunk 1]
```diff
@@ -1,8 +1,3 @@
-#!/bin/bash
-#set -e
+#!/bin/env zsh
```
Comment (guidoiaquinti) [inline .devcontainer/container_start.sh:1]: is `zsh` the default shell?

File: .devcontainer/docker-compose.yml [hunk 2]
```diff
@@ -1,37 +1,85 @@
...
+    # I expect these to be run from within this workspace context, such that you
+    # do not need to interact with them via docker, the intent is to make them
+    # readily debuggable.
+    #
+    # Service dependencies are, however, run separately as there isn't so much
+    # of a need to debug them. You can of course however use `docker logs` (as we
+    # have docker-in-docker enabled) to view these, of via the vscode docker
+    # extention.
+    app:
+        image: ghcr.io/posthog/posthog/codespaces:master
```
Comment (guidoiaquinti) [inline .devcontainer/docker-compose.yml:20]: why not pushing directly to DockerHub (like we do with other images)?

File: .github/workflows/codespaces.yml [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+            # As ghcr.io complains if the image has upper case letters, we use
+            # this action to ensure we get a lower case version. See
+            # https://github.com/docker/build-push-action/issues/237#issuecomment-848673650
+            # for more details
+            - name: Docker image metadata
+              id: meta
+              uses: docker/metadata-action@v3
+              with:
+                  images: ghcr.io/${{ github.repository }}/codespaces
```
Comment (guidoiaquinti) [inline .github/workflows/codespaces.yml:42]: ditto

File: .devcontainer/container_start.sh (hunk 1 above)
Comment (hazzadous) [reply .devcontainer/container_start.sh:1]: Yes, is the default for [this image](https://github.com/PostHog/posthog/pull/7234/files#diff-13bd9d7a30bf46656bc81f1ad5b908a627f9247be3f7d76df862b0578b534fc6R12)

File: .devcontainer/docker-compose.yml (hunk 2 above)
Comment (hazzadous) [reply .devcontainer/docker-compose.yml:20]: These images I don't think are relevant outside of github, and github has tooling around ghcr, so thought I'd lean into github a bit for this. Do you think worth changing?

File: .github/workflows/codespaces.yml (hunk 3 above)
Comment (hazzadous) [reply .github/workflows/codespaces.yml:42]: Yup same, they're only relevant within github so thought I'd just check into ghcr, without polluting the public facing dockerhub page.

**Pull request merged**